src/array/attribute.rs

```diff
@@ -234,6 +235,67 @@ impl Attribute {
             Err(ctx.expect_last_error())
         }
     }
+
+    // This currently does not support setting multi-value cells.
+    pub fn set_fill_value<Conv: CAPIConverter + 'static>(
+        &self,
+        ctx: &Context,
+        value: Conv,
+    ) -> TileDBResult<()> {
+        let c_val: Conv::CAPIType = value.to_capi();
+
+        if !self.datatype(ctx)?.is_valid(&value) {
+            return Err(Error::from(format!(
+                "Dimension type mismatch: expected {}, found {}",
+                self.datatype(ctx)?,
+                std::any::type_name::<Conv>()
+            )));
+        }
+
+        let res = unsafe {
+            ffi::tiledb_attribute_set_fill_value(
+                ctx.as_mut_ptr(),
+                *self.raw,
+                &c_val as *const Conv::CAPIType as *const std::ffi::c_void,
+                std::mem::size_of::<Conv::CAPIType>() as u64,
+            )
+        };
+
+        if res == ffi::TILEDB_OK {
+            Ok(())
+        } else {
+            Err(ctx.expect_last_error())
+        }
+    }
+
+    pub fn get_fill_value<Conv: CAPIConverter>(
+        &self,
+        ctx: &Context,
+    ) -> TileDBResult<Conv> {
+        let mut c_ptr: *const std::ffi::c_void = out_ptr!();
+        let mut c_size: u64 = 0;
+
+        let res = unsafe {
+            ffi::tiledb_attribute_get_fill_value(
```

Comment:
I don't want to go too far Rust-ifying our interface until more of the library is wrapped so that we have a better feel for common patterns, but I wonder if we couldn't do something along the lines of having a specific type for the value would be the way to go. Something along the lines of:

```
enum FillValue {
    NonNullable(T),
    Nullable(T, B)
}

fun set_fill_value(value: FillValue);
fun get_fill_value() -> FillValule;
```